.github/workflows/changelog_update.yml

```
@@ -0,0 +1,38 @@
+      - uses: actions/checkout@v4
+      - name: Set up Ruby
+        uses: ruby/setup-ruby@v1
+        with:
+          ruby-version: '3.2.2'
+          bundler-cache: true
+      - name: Configure Git
+        run: |
+          git config user.name "GitHubActions"
+          git config user.email "<>"
+      - name: Generate and push changelog
+        run: |
```

actionlint (1.7.4)

shellcheck reported issue in this script: SC2086:info:1:90: Double quote to prevent globbing and word splitting

(shellcheck)